peps/pep-0735.rst

```diff
@@ -111,6 +113,20 @@ downsides:
   concerned about ensuring that their development extras are not confused with
   user-facing extras.
 
+Use of ``[project.optional-dependencies]`` outside of packages
+''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+Although some tools (e.g., ``uv``) define direct use of the
+``[project.optional-dependencies]`` table, this is distinct from the usage of
+this table defined in :pep:`621`, and should be better understood as a direct
+read of TOML data from ``pyproject.toml``.
+
+Under :pep:`621`, use of the ``[project]`` table stipulates required fields
+(``name`` and ``version``). Therefore, using
+``[project.optional-dependencies]`` without specifying these fields means that
+a project is not defining :pep:`621` metadata. This may or may not be allowed,
+and in particular many tools will treat partial package metadata as an error.
```

Comment:
I'm not sure I follow along exactly on this paragraph, but uv does ban `[project]` table without name and version, e.g. the following errors:
```toml
[project]
optional-dependencies = {}
```

Maybe this is referring to uv allowing you to define `[project]` table without a `[build-system]` entry and then doesn't build the project, to support the users who want to manage the dependencies without ever producing a wheel?

Reply:
I think this means that I didn't properly understand what the `uv` devs were trying to say in the discourse thread. I thought they were saying they allowed for `[project]` sans the `name` and `version` fields -- that's been a key part of a lot of the past discussion about "projects which don't build a wheel" -- and I found that possibility disconcerting but didn't know what to make of it.

Having a specialized behavior when `[build-system]` is absent is much more subtle.

I was already tending towards removing this, so I think this puts it over the top. I'll also circle back to the discourse thread to note it. Thanks for catching this error!